packages/react/src/index.d.ts

```diff
@@ -0,0 +1,1162 @@
+    aspectRatio?: 'square' | 'classic' | 'wide' | 'cinema';
+    src?: string;
+    width?: number;
+    height?: number;
+  }
+  export const Media: React.FC<MediaProps>;
+
+  export interface MessageProps extends BasicProps {
+    type?: 'form' | 'system';
+    variant?: 'information' | 'positive' | 'warning' | 'negative';
+    dismissible?: boolean;
+    show?: boolean;
+    onClose?: () => void;
+  }
+  export const Message: React.FC<MessageProps> & {
+    Content: React.FC<BasicWithAsProps>;
+    Container: React.FC<BasicWithAsProps>;
+    Title: React.FC<BasicWithAsProps>;
+  };
+
+  export interface ModalHeaderProps extends BasicProps {
+    onHide?: () => void;
+    closeButton?: boolean;
+  }
+  export interface ModalProps extends BasicProps {
+    size?: 'small' | 'medium' | 'large' | 'extraLarge';
+    relative?: boolean;
+    centered?: boolean;
+    show?: boolean;
+    onHide?: () => void;
+  }
+  export const Modal: React.FC<ModalProps> & {
+    Title: React.FC<ModalHeaderProps>;
+    Body: React.FC<BasicProps>;
+    Footer: React.FC<BasicProps>;
+    Header: React.FC<ModalHeaderProps>;
+    Inside: React.FC<BasicProps>;
+  };
+
+  export interface MultiStepsItemProps extends BasicProps {
+    isLast?: boolean;
+    isCompleted?: boolean;
+    isActive?: boolean;
+    title?: string;
+    disabled?: boolean;
+  }
+  export interface MultiStepsProps extends BasicProps {
+    current?: number;
+    currentLabel?: string;
+    onChange?: (current: number) => void;
+    direction?: 'horizontal' | 'vertical';
+    variant?:
+      | 'primary'
+      | 'accent'
+      | 'positive'
+      | 'warning'
+      | 'negative'
+      | 'white';
+  }
+  export const MultiSteps: React.FC<MultiStepsProps> & {
+    Item: React.FC<MultiStepsItemProps>;
+  };
+
+  interface OverlayBasicProps extends BasicProps {
+    children: React.ReactNode;
+    popperConfig?: PopperOptions;
+    placement?: PopperPlacement;
+    rootClose?: boolean;
+    rootCloseEvent?: string;
+    rootCloseDisabled?: boolean;
+  }
+  export interface OverlayTriggerProps extends OverlayBasicProps {
+    trigger?: TriggerType | TriggerType[];
+    delay?: number | { show: number; hide: number };
+    hoverOverlay?: boolean;
+    defaultShow?: boolean;
+    overlay: FuncType | React.ReactNode;
+  }
+
+  export interface OverlayProps extends OverlayBasicProps {
+    show?: boolean;
+    target?: React.RefObject<HTMLElement>;
+    container?: HTMLElement;
+    flip?: boolean;
+    containerPadding?: number;
+    onHide?: () => void;
+  }
+  export const Overlay: React.FC<OverlayProps> & {
+    Trigger: React.FC<OverlayTriggerProps>;
+  };
+
+  export interface PageLayoutProps extends BasicWithAsProps {
+    headerProps?: object;
+    footerProps?: object;
+    bodyProps?: object;
+  }
+  export const PageLayout: React.FC<PageLayoutProps> & {
+    Header: React.FC<BasicWithAsProps>;
+    Footer: React.FC<BasicWithAsProps>;
+    Body: React.FC<BasicWithAsProps>;
+  };
```

Comment:
I think they only extends `BasicProps` right, since there is no `prop` as in these component

Reply:
It created from `ultils/createBlock` and has prop `as` https://github.com/tinhvqbk/ahaui/blob/e957aca434578f10a806921a6d065b4fe7393489/packages/react/src/utils/createBlock.js#L12